package/src/components/MessageList/MessageList.tsx

```diff
@@ -756,20 +753,20 @@ const MessageListWithContext = (props: MessageListPropsWithContext) => {
       const showUnreadUnderlay = !!shouldShowUnreadUnderlay && showUnreadSeparator;
 
       const wrapMessageInTheme = client.userID === message.user?.id && !!myMessageTheme;
-      const renderDateSeperator = isMessageWithStylesReadByAndDateSeparator(message) &&
-        message.dateSeparator && <InlineDateSeparator date={message.dateSeparator} />;
+      const renderDateSeperator = dateSeparators[message.id] && (
```

Comment:
Not having `dateSeparator` on our `message` is now very breaking in terms of the fact integrators will no longer have a way of using this. The general idea was to remove the enriched properties from `message`, but provide an alternative way to get them (and use them only where they're needed - for example, `readBy` is really only used in `MessageStatus` so there's no need for us to pass it to the entire `message`).